superset-frontend/src/common/components/Modal.tsx

```diff
@@ -121,12 +137,15 @@ export default function Modal({
       ]
     : footer;
 
+  const modalWidth = width || (responsive ? 'calc(100vw - 24px)' : '600px');
```

Comment:
NIT: This works, but it might be nice to do it in the `StyledModal` CSS so it can use `gridUnit` rather than use the `width` prop on line 146

Reply:
I changed it to `responsive ? '100vw' : '600px'` and used `padding: 0 ${theme.gridUnits * 3}px` in styled. The reason for that is that Antd Modal uses default value for width and puts it in `style`, so I'd need to use `!important` to override it.